src/Framework/NativeMethods.cs

```diff
@@ -1480,6 +1480,69 @@ internal static void VerifyThrowWin32Result(int result)
         }
     }
 
+#if !CLR2COMPATIBILITY
+    internal static (bool acceptAnsiColorCodes, bool outputIsScreen, uint? originalConsoleMode) QueryIsScreenAndTryEnableAnsiColorCodes()
+    {
+        bool acceptAnsiColorCodes = false;
+        bool outputIsScreen = false;
+        uint? originalConsoleMode = null;
+
+        if (!Console.IsOutputRedirected)
```

Comment:
minor: Could the if block hierarchy be flattened?

```
if (!Console.IsOutputRedirected && 
IsWindows &&
GetStdHandle(STD_OUTPUT_HANDLE) is IntPtr stdOut &&
GetConsoleMode(stdOut, out uint cnosoleMode){...}
```

Reply:
Doesn't seem to be possible to go this far. I have removed one level of nesting.